pkg/lib/oauth/oidc/id_token.go

```diff
@@ -239,7 +239,7 @@ func (ti *IDTokenIssuer) PopulateUserClaims(token jwt.Token, userID string, nonP
 	return nil
 }
 
-func (ti *IDTokenIssuer) GetUserInfo(userID string) (map[string]interface{}, error) {
+func (ti *IDTokenIssuer) GetUserInfo(userID string, client *config.OAuthClientConfig, scopes []string) (map[string]interface{}, error) {
```

Comment:
Maybe we can introduce a data structure to represent the following states

1. IDP session
2. First party client
3. Third party client

```golang
type ClientParty string
const ClientPartyFirst ClientParty = "first_party"
const clientPartyThird ClientParty = "third_party"

type ClientLike struct {
  ClientParty ClientParty
  Scopes []string
}

var ClientLikeIDPSession = &ClientLike{
  ClientParty: ClientPartyFirst,
  Scopes: []string{FullAccessScope},
}

var ClientLikeNotFound = &ClientLike{
  ClientParty: ClientPartyThird,
}

func (s *IDPSession) ClientLike() *ClientLike {
  return ClientLikeIDPSession
}

func (s *OfflineGrant) ClientLike(c *config.OAuthConfig) *ClientLike {
  client, ok := c.GetClient(s.ClientID)
  if !ok {
    return ClientLikeNotFound
  }
  return &ClientLike{
    ClientParty: client.ClientParty(),
    Scopes: s.Scopes,
  }
}
```

We can remove `IsFirstParty()` and use `ClientParty()` instead.

GetUserInfo should takes `(userID string, client *ClientLike)`

Reply:
Rewrite the commits to introduce `ClientLike`, and see if it looks better!